### PR TITLE
Add list of providers to /browse page with pagination and search by city

### PR DESCRIPTION
**Pull request by kpennachio**
First pass at list of providers on the /browse page. (If we need this to be all or nothing lmk!)
* Paginated list of providers with next/previous page buttons
* Search by city. This dynamically searches as you type for case insensitive string inclusion.
* Links to provider pages.

This page is still missing:
* Map and distance away (once we are using Google Places API?)
* More styling to match Figma
* More detailed page options (able to click on another page number vs just forward and back)

Screen Shot 2020-03-28 at 5 15 33 PM

### CODE REVIEW

File: app/graphql/types/provider.rb [hunk 1]
```diff
@@ -9,6 +9,7 @@ class Provider < Types::BaseObject
     field :state, String, null: false
     field :description, String, null: false
     field :requests, [Types::Request], null: false
+    field :responses, Types::Connections::ResponseConnection, connection: true, null: true
```
Comment (cantino) [inline app/graphql/types/provider.rb]: It looks like you don't need the ResponseConnection or ResponseEdge to be added if you're only looking for the count. Instead, let's do:

```
field :response_count, Integer, null: false
def response_count
  object.responses.count
end
```

File: app/graphql/types/query_type.rb [hunk 2]
```diff
@@ -5,18 +5,30 @@ class QueryType < Types::BaseObject
...
+      active_requests = filters.delete(:active_requests)
+      city = filters.delete(:city)
       scope = ::Provider.where(where(filters)).order(order(order_by))
+
       if updated_within_days
         scope = scope.where("updated_at > ?", updated_within_days.days.ago)
       end
+
+      if active_requests
+        scope = scope.where("requests @> ?", '[{"satisfied": false}]')
```
Comment (cantino) [inline app/graphql/types/query_type.rb:25]: This is fine for now— and a good trick!— but we may want to add a cache column on Provider that we maintain if this gets slow.

**Review by cantino — CHANGES_REQUESTED** [review]
Awesome, nice works! Let's not link this yet, but we can merge it soon and iterate on it.

File: app/graphql/types/provider.rb (hunk 1 above)
Comment (kpennachio) [reply app/graphql/types/provider.rb]: Ah sounds good! I'll make the change.

**Comment by kpennachio** [comment]
@cantino do you have a preferred "unlinked" url for this page in progress or should I just keep the files? I'll revert back to what was previously showing on /browse for now.

**Comment by cantino** [comment]
> @cantino do you have a preferred "unlinked" url for this page in progress or should I just keep the files? I'll revert back to what was previously showing on /browse for now.

You don't need to revert, since we don't link to /browse yet anyway, I think you can put it there?

**Comment by kpennachio** [comment]
Located at /browse-new for iteration. Thanks for the review @cantino !

**Review by cantino — APPROVED** [review]

**Pull request merged**